src/app/GitCommands/Settings/EffectiveGitConfigSettings.cs

```diff
@@ -0,0 +1,37 @@
+#nullable enable
+
+using System.Diagnostics;
+using GitExtensions.Extensibility;
+using GitExtensions.Extensibility.Configurations;
+using GitExtensions.Extensibility.Git;
+
+namespace GitCommands.Settings;
+
+[DebuggerDisplay("{" + nameof(DebuggerDisplay) + ",nq}")]
+public sealed class EffectiveGitConfigSettings(IExecutable gitExecutable) : GitConfigSettingsBase(gitExecutable, GitSettingLevel.Effective), IConfigValueStore
```

Comment:
Does this and the rest of the hierarchy need to be public?
Would be great to have some xml-docs for the types and public members

Reply:
`public` because used by GUI project.
I have added xml-docs to classes and to `protected` stuff.
Public methods are interface implementations and inherit the xml-docs.